src/os/bluestore/BlueStore.cc

```diff
@@ -3762,6 +3762,12 @@ void *BlueStore::MempoolThread::entry()
 
   bool interval_stats_trim = false;
   while (!stop) {
+    // Update pcm cache settings if osd_memory_target has been changed
+    if (store->osd_memory_target != target && pcm != nullptr) {
```

Comment:
IMO since pcm is used inside _update_cache_settings() it's better to check it for nullptr inside..

Reply:
Done.